src/test/ui/closures/2229_closure_analysis/diagnostics/cant-mutate-imm-borrow.rs

```diff
@@ -14,7 +14,6 @@ fn main() {
     let mut c = || {
     //~^ ERROR: cannot borrow `z.0.0.0` as mutable, as it is behind a `&` reference
         z.0.0.0 = format!("X1");
```

Comment:
I would have expected us to not try read the place `z` on its own. When the feature isn't enabled we only read `z.0.0.0`.

Debug logs from the report_mutability_error when the feature is enabled.
```
DEBUG rustc_mir::borrow_check::diagnostics::mutability_errors report_mutability_error: access_place_desc=Some("z")
....
DEBUG rustc_mir::borrow_check::diagnostics::mutability_errors report_mutability_error: access_place_desc=Some("z.0.0.0")
```

Reply:
This is definitely not the error message we want-- we'd like the error to be reported on `z.0.0.0`, right? I think this may be related to my comment about the `used_mut_upvars` but I'm not 100% sure.